apps/extension/src/approve-origin.ts

```diff
@@ -4,13 +4,16 @@ import { errorFromJson } from '@connectrpc/connect/protocol-connect';
 import { localExtStorage } from '@penumbra-zone/storage';
 import { OriginApproval, PopupType } from './message/popup';
 import { popup } from './popup';
+import Map from '@penumbra-zone/polyfills/Map.groupBy';
+import { UserAttitude } from '@penumbra-zone/types/src/user-attitude';
 
 export const originAlreadyApproved = async (url: string): Promise<boolean> => {
   // parses the origin and returns a consistent format
   const urlOrigin = new URL(url).origin;
   // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  const connectedSites = await localExtStorage.get('connectedSites');
-  return Boolean(connectedSites[urlOrigin]);
+  const knownSites = await localExtStorage.get('knownSites');
+  const existingRecord = knownSites.find(site => site.origin === urlOrigin);
+  return existingRecord?.attitude === UserAttitude.Approved;
 };
 
 export const approveOrigin = async ({
```

Comment:
This does not seem to be responsive if the user is logged out. If it's beyond the scope of this PR, can you open an issue for this?

Reply:
not sure what this is referring to but it works fine for me
https://github.com/penumbra-zone/web/assets/134443988/b24522cd-4bff-4905-9052-52f33b41a173

Reply:
Strange, doesn't block merge but perhaps @jessepinho you can test this out after it hits main